src/backend/utils/time/snapmgr.c

```diff
@@ -2245,6 +2250,47 @@ RestoreTransactionSnapshot(Snapshot snapshot, void *source_pgproc)
 
 /*
  * XidInMVCCSnapshot
+ *
+ * Check whether this xid is in snapshot. When enable_csn_snapshot is
+ * switched off just call XidInLocalMVCCSnapshot().
+ */
+bool
+XidInMVCCSnapshot(TransactionId xid, Snapshot snapshot)
+{
+	bool in_snapshot;
+
+	in_snapshot = XidInLocalMVCCSnapshot(xid, snapshot);
```

Comment:
So we still use the "local" mechanism, too, when CSN snapshots are enabled. Why do we need both? If the "local" snapshot says the transaction is still in progress, but the CSN says it's not, or vice versa, which is it?